src/Raven.Server/ServerWide/ClusterStateMachine.License.cs

```diff
@@ -0,0 +1,207 @@
+
+            case nameof(AddSqlEtlCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion54116) == false)
+                    return;
+
+                if (serverStore.LicenseManager.LicenseStatus.HasSqlEtl)
+                    return;
+
+                throw new LicenseLimitException(LimitType.SqlEtl, "Your license doesn't support adding SQL ETL feature.");
+
+            case nameof(EditTimeSeriesConfigurationCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion54116) == false)
+                    return;
+
+                if (serverStore.LicenseManager.LicenseStatus.HasTimeSeriesRollupsAndRetention)
+                    return;
+
+                if (databaseRecord.TimeSeries.Collections.Count > 0 && databaseRecord.TimeSeries.Collections.All(x => x.Value.Disabled))
+                    return;
+
+                throw new LicenseLimitException(LimitType.TimeSeriesRollupsAndRetention, "Your license doesn't support adding Time Series Rollups And Retention feature.");
+
+            case nameof(EditDocumentsCompressionCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion54116) == false)
+                    return;
+
+                if (serverStore.LicenseManager.LicenseStatus.HasDocumentsCompression)
+                    return;
+
+                if (databaseRecord.DocumentsCompression.CompressAllCollections == false && databaseRecord.DocumentsCompression.CompressRevisions == false)
+                    return;
+
+                throw new LicenseLimitException(LimitType.DocumentsCompression, "Your license doesn't support adding Documents Compression feature.");
+
+            case nameof(AddElasticSearchEtlCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion54116) == false)
+                    return;
+
+                if (serverStore.LicenseManager.LicenseStatus.HasElasticSearchEtl)
+                    return;
+
+                throw new LicenseLimitException(LimitType.ElasticSearchEtl, "Your license doesn't support adding Elastic Search ETL feature.");
+
+            case nameof(AddOlapEtlCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion54116) == false)
+                    return;
+
+                if (serverStore.LicenseManager.LicenseStatus.HasOlapEtl)
+                    return;
+
+                throw new LicenseLimitException(LimitType.OlapEtl, "Your license doesn't support adding Olap ETL feature.");
+
+            case nameof(AddQueueEtlCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion54116) == false)
+                    return;
+
+                if (serverStore.LicenseManager.LicenseStatus.HasQueueEtl)
+                    return;
+
+                throw new LicenseLimitException(LimitType.QueueEtl, "Your license doesn't support adding Queue ETL feature.");
```

Comment:
I see in 6.0 we have a dedicated method `LicenseManager,GenerateLicenseLimit()` shouldn't we have similar concept in here?

Reply:
I don't think this is necessary. we only throw and don't need to add warnings and other info about the license like at GenerateLicenseLimit()

Reply:
OK